mne/epochs.py

```diff
@@ -3543,31 +3544,34 @@ def equalize_epoch_counts(epochs_list, method="mintime"):
     --------
     >>> equalize_epoch_counts([epochs1, epochs2])  # doctest: +SKIP
     """
-    if not all(isinstance(e, BaseEpochs) for e in epochs_list):
+    if not all(isinstance(epoch, (BaseEpochs, EpochsTFR)) for epoch in epochs_list):
         raise ValueError("All inputs must be Epochs instances")
 
     # make sure bad epochs are dropped
-    for e in epochs_list:
-        if not e._bad_dropped:
-            e.drop_bad()
-    event_times = [e.events[:, 0] for e in epochs_list]
-    indices = _get_drop_indices(event_times, method)
-    for e, inds in zip(epochs_list, indices):
-        e.drop(inds, reason="EQUALIZED_COUNT")
-
-
-def _get_drop_indices(event_times, method):
+    for epoch in epochs_list:
+        # for EpochsTFR we can set _bad_dropped to True
+        if isinstance(epoch, EpochsTFR):
+            epoch._bad_dropped = True
+        if not epoch._bad_dropped:
+            epoch.drop_bad()
+    event_indices = [epoch.events[:, 0] for epoch in epochs_list]
+    indices = _get_drop_indices(event_indices, method)
```

Comment:
you're changing this variable name from `event_times` to `event_indices`...  the original name was not accurate but the new name isn't either. Let's call it `sample_nums` (since the first column of `events` is the sample number)

```suggestion
    sample_nums = [epoch.events[:, 0] for epoch in epochs_list]
    indices = _get_drop_indices(sample_nums, method)
```